### PR TITLE
Peter Bowman solution in python done

### PR DESCRIPTION
**Pull request by bowmanpete123**

### CODE REVIEW

**Comment by den-t** [comment]
Hi Peter.
It's better organised than the original but essentially has the same approach.
There's an unused enum import. So I suspect there was more to it, just haven't made it to the merge request?
For pretty much any solution we'd want to encapsulate each check into a separate function/routine. So required checks could be supplied by the means of configuration or similar. There a part "reuse the validator without applying max length restriction, and include another rule" which requires the validation to be more flexible about which checks to apply.
The pattern that could help is [here](https://en.wikipedia.org/wiki/Chain-of-responsibility_pattern#Python_Example)

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (MatiasIac) [inline .vscode/settings.json:1]: Make sure to do not include, in the commits, files created by the IDE.

**Comment by bowmanpete123** [comment]
Ahk; I'll have another go at it and see what I exactly committed; git on my machine was having problems getting it up to github so I'll head back and make sure I've got it all sorted.

**Comment by kellymarjorie** [comment]
Closing to tidy up the list :)

**Pull request closed without merging**